Extend the Beman Standard: add release related rules (#108)
bemanproject/beman: docs/BEMAN_STANDARD.md

Known exceptions:
* [mpark/wg21: Framework for Writing C++ Committee Papers](https://github.com/mpark/wg21): A non-C++ submodule designed for drafting ISO C++ papers using LaTeX or Markdown.

## Release

> **[RELEASE_GITHUB]**: REQUIREMENT: All Beman libraries must be released using [GitHub Releases](https://docs.github.com/en/repositories/releasing-projects-on-github).

Author: ```suggestion
**[RELEASE.GITHUB]**: REQUIREMENT: All Beman libraries must be released using [GitHub Releases](https://docs.github.com/en/repositories/releasing-projects-on-github).
```

> **[RELEASE_NOTES]**: RECOMMENDATION: Release should include:

Author: ```suggestion
**[RELEASE.NOTES]**: RECOMMENDATION: Release should include:
```

> **[RELEASE_NOTES]**: REQUIREMENT: Release notes must include:

Author: ```suggestion
**[RELEASE_NOTES]**: RECOMMENDATION: Release should include:
```

1. A title that clearly indicates the release version and purpose.
2. An "About" section describing the main purpose and impact of the release.
3. A "Release Maturity" section indicating the current status of the library according to the [Beman Library Maturity Model](https://github.com/bemanproject/beman/blob/main/docs/BEMAN_LIBRARY_MATURITY_MODEL.md).

> 4. A "What's Changed" section listing all changes, organized by:

Reviewer: Can you please a few lines with what should be added for each section? Maybe we also could create a template to use.

Author: I already provided a template.

> 4. A "What's Changed" section listing all changes, organized by:

Author: ```suggestion
4. A "Changelog" section listing all changes, organized by:
```

   - Breaking changes (if any)
   - New features and improvements
   - Bug fixes
   - Build system changes
   - Documentation updates
5. A "New Contributors" section listing all first-time contributors.
6. A "Full Changelog" link to the complete list of commits.
7. A "Contributors" section listing all contributors to this release. Note: This section is autogenerated by GitHub Releases!

Use the following format:

> ```markdown

Reviewer: Should we also had a placeholder for the title?

Author: No. Updated the docs (`Check visual example for the field location.`) and the visual example.

## About

<description ... >

## Release Maturity

This release has a status of [Under development and not yet ready for production use.](https://github.com/bemanproject/beman/blob/main/docs/BEMAN_LIBRARY_MATURITY_MODEL.md#under-development-and-not-yet-ready-for-production-use).

> ## What's Changed

Author: ```suggestion
## Changelog
```

> ## What's Changed

Reviewer: can we name this to changelog?

Author: done

* <description change> by @${USER} in https://github.com/bemanproject/${REPO}/pull/${PR_ID}
* ...

## New Contributors

* @${USER} made their first contribution in https://github.com/bemanproject/${REPO}/pull/#${PR_ID}
* ...

**Full Changelog**: https://github.com/bemanproject/${REPO}/compare/${PREV_VERSION}...${CURRENT_VERSION}
```

Complete examples can be found in [https://github.com/bemanproject/exemplar/releases](https://github.com/bemanproject/exemplar/releases).

> 

Author: ```suggestion

Here is a visualization of a release in exemplar: 
<img width="1722" alt="image" src="https://github.com/user-attachments/assets/785ddf04-0ac9-41f8-acc8-d5a49c9897a9" />

```

Use the following format:

```markdown
## About

<!-- TODO: Update this section and remove this comment. -->

## Release Maturity

<!-- TODO: Update this section and remove this comment. -->

This release has a status of [Under development and not yet ready for production use.](https://github.com/bemanproject/beman/blob/main/docs/BEMAN_LIBRARY_MATURITY_MODEL.md#under-development-and-not-yet-ready-for-production-use).

## Changelog

<!-- TODO: Update this section and remove this comment. -->

* <description change> by @${USER} in https://github.com/bemanproject/${REPO}/pull/${PR_ID}
* ...

**Full Changelog**: https://github.com/bemanproject/${REPO}/compare/${PREV_VERSION}...${CURRENT_VERSION}

## New Contributors

<!-- TODO: Update this section and remove this comment. -->

* @${USER} made their first contribution in https://github.com/bemanproject/${REPO}/pull/#${PR_ID}
* ...

```

Complete examples can be found in [https://github.com/bemanproject/exemplar/releases](https://github.com/bemanproject/exemplar/releases).

Here is a visualization of a release in exemplar: 

> <img width="1722" alt="image" src="https://github.com/user-attachments/assets/785ddf04-0ac9-41f8-acc8-d5a49c9897a9" />

Author: Same observation as in https://github.com/bemanproject/beman/pull/109#discussion_r2033116385. I will add a static PNG files into `docs/img` and render "local" image (from repo).
